dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/icon/JdbcIconStore.java

```diff
@@ -210,6 +215,15 @@ private String buildIconQuery(
       parameterSource.addValue("keys", params.getKeys());
     }
 
+    if (params.hasSearch()) {
+      String searchValue = params.getSearch();
+
+      sql += hlp.whereAnd() + " c.iconkey ilike :search";
+      sql += " OR jsonb_path_exists(keywords, '$[*] ? (@ like_regex " + doubleQuote(searchValue) + ")')";
```

Comment:
I had ALOT of problems getting the correct escape params to work here. Whatever I tried I was not able to use the parameter source to put in the searchValue in the `jsonb` query.

I did intent to add start-of-string anchor to this, but I see I forgot about that. Any thoughts about that?

Reply:
Alright, I managed to add the parameter using `parameterSource`. But I had to add doubleQuotes in the SQL itself for it to work.